### DISCUSSION TITLE
Keycloak IDP with same keycloak instance

### DISCUSSION THREAD

**Discussion by althaf004** [keycloak/keycloak#27115] [Q&A]
Hi,

We've been working on a requirement to allow a specific set of users to access all realms within a Keycloak instance using the same username and password. This setup ensures that a password change reflects across all realms seamlessly.

Here's what we've done so far:

1) Created users and credentials in the master realm along with a client.
2) In the user realm (ABC), we set up an identity provider using the client details from the master realm.
3) Replicated the user accounts in the user realm (ABC) with no credentials.
4) This configuration has enabled a login button in the user realm (ABC), allowing users to log in successfully. However, we're encountering an issue where the following error pops up intermittently:

![image](https://github.com/keycloak/keycloak/assets/83403907/d78a2bc0-1694-4a22-bc2f-6b7440349277)

Interestingly, this problem doesn't occur when using a private browser.

We're seeking guidance on how to resolve this issue effectively.

Thanks

**Comment by vmuzikar** [comment]
Is there anything in the log hinting what went wrong when the error happens?

**Reply by althaf004** [reply]
In network, got 502 Bad Gateway error.

**Reply by vmuzikar** [reply]
Can you share the full stack trace?
